# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 7b00a16ad21e
# Full Hash: 7b00a16ad21ee2b820b679199ae3806250d861e9
# Author: Nathan LaPre <nlapre@mozilla.com>
# Date: 2024-09-10 09:03:27
# Regressor Bug: 1794974
# File Overlap Count: 1
# Description:
#   Bug 1794974: Part 2: Add ability to query and send individual cache domains, r=Jamie
#   
#   This revision adds the capability of querying and sending information about
#   individual cache domains. It introduces the concept of active cache domains to
#   the accessibility service: cache domains that we think clients need. Conversely,
# ==============================================================================

diff -r 642b6d8c4d3b -r 7b00a16ad21e accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Mon Sep 09 23:02:19 2024 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Mon Sep 09 23:02:19 2024 +0000
@@ -3337,6 +3337,15 @@
     return;
   }
 
+  // Only send cache updates for domains that are active.
+  const uint64_t domainsToSend =
+      nsAccessibilityService::GetActiveCacheDomains() & aCacheDomain;
+
+  // Avoid sending cache updates if we have no domains to update.
+  if (domainsToSend == CacheDomain::None) {
+    return;
+  }
+
   DocAccessibleChild* ipcDoc = mDoc->IPCDoc();
   if (!ipcDoc) {
     // This means DocAccessible::DoInitialUpdate hasn't been called yet, which
@@ -3347,7 +3356,7 @@
   }
 
   RefPtr<AccAttributes> fields =
-      BundleFieldsForCache(aCacheDomain, aUpdateType);
+      BundleFieldsForCache(domainsToSend, aUpdateType);
   if (!fields->Count()) {
     return;
   }
@@ -3369,7 +3378,8 @@
 }
 
 already_AddRefed<AccAttributes> LocalAccessible::BundleFieldsForCache(
-    uint64_t aCacheDomain, CacheUpdateType aUpdateType) {
+    uint64_t aCacheDomain, CacheUpdateType aUpdateType,
+    uint64_t aInitialDomains) {
   RefPtr<AccAttributes> fields = new AccAttributes();
 
   // Caching name for text leaf Accessibles is redundant, since their name is